pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -290,6 +291,7 @@ func newOpenStack(cfg Config) (*OpenStack, error) {
 		return nil, err
 	}
 
+	provider.HTTPClient.Timeout = cfg.Metadata.RequestTimeout
```

Comment:
@dixudx - we should set also default timeout value if none provided.

Reply:
In gophercloud, if no value is set, the default value is zero which means no timeout.